modules/core/src/test/java/org/apache/ignite/cdc/AbstractCdcTest.java

```diff
@@ -392,6 +412,50 @@ public static class UserCdcConsumer extends TestCdcConsumer<Integer> {
         }
     }
 
+    /** */
+    public static class TrackCacheEventsConsumer implements CdcConsumer {
+        /** Cache events. */
+        public Map<Integer, CdcCacheEvent> evts = new ConcurrentHashMap<>();
+
+        /** {@inheritDoc} */
+        @Override public void onCacheChange(Iterator<CdcCacheEvent> cacheEvents) {
+            cacheEvents.forEachRemaining(e -> evts.put(e.cacheId(), e));
+        }
+
+        /** {@inheritDoc} */
+        @Override public void onCacheDestroy(Iterator<Integer> caches) {
+            caches.forEachRemaining(evts::remove);
```

Comment:
Lets add a check that cache was exist/do not removed twice

Reply:
This can't be done - because cache events received by consumer on each node. So, multiple cache events will be received for single cache operation.